src/virtio/src/console/virtio/pty_handler.rs

```diff
@@ -0,0 +1,98 @@
+use std::fs::{File, OpenOptions};
+use std::io::{Read, Write};
+use std::os::fd::AsRawFd;
+use std::os::unix::fs::OpenOptionsExt;
+use std::sync::{Arc, Mutex};
+
+use event_manager::{EventOps, Events, MutEventSubscriber};
+use libc::IN_NONBLOCK;
+use std::os::unix::net::UnixStream;
+use virtio_console::console::Console;
+use vm_memory::WriteVolatile;
+use vmm_sys_util::epoll::EventSet;
+use vmm_sys_util::eventfd::EventFd;
+
+const SOURCE_PTY: u32 = 0;
+const SOURCE_SOCKET: u32 = 1;
+
+const BUFFER_SIZE: usize = 128;
+
+pub(super) struct PtyHandler<W: Write + WriteVolatile> {
+    pub pty: File,
+    pub socket: UnixStream,
+    pub console: Arc<Mutex<Console<W>>>,
+    pub input_ioeventfd: EventFd,
+}
+
+impl<W> PtyHandler<W>
+where
+    W: Write + WriteVolatile,
+{
+    pub fn new(
+        socket: UnixStream,
+        console: Arc<Mutex<Console<W>>>,
+        input_ioeventfd: EventFd,
+    ) -> Self {
+
+        let pty = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(true)
+            .custom_flags(IN_NONBLOCK)
+            .open("/dev/ptmx").unwrap();
+
+        let name = unsafe {
+            libc::grantpt(pty.as_raw_fd());
+            libc::unlockpt(pty.as_raw_fd());
+            std::ffi::CStr::from_ptr(libc::ptsname(pty.as_raw_fd()))
+        };
+
+        println!("virtio-console at {}", name.to_str().unwrap());
```

Comment:
Include the device model ID in the print message to support multiple VirtIO consoles.
Rather than relying on the print order, adding the device model ID provides a clearer identification of each console.

Reply:
Done in the latest push.  I've also added a config option `pty_alias` so the user can configure a custom symlink for the created pty.